nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -373,11 +372,16 @@ let
           ''}
         '';
 
+        # sendfile works well with kTLS/SSL_sendfile() from openssl but not with http2
+        # https://mailman.nginx.org/pipermail/nginx-devel/2021-December/014601.html
+        # https://mailman.nginx.org/pipermail/nginx-devel/2021-December/014603.html
+        sendfile = lib.optionalString (cfg.recommendedOptimisation && !(hasSSL && vhost.http2)) "sendfile on;";
```

Comment:
Maybe rename it to `enableSendfile`? To avoid confusion with the options that nginx uses.